### PR TITLE
fixes jazzbar table lock/unlock functionality

### PR DESCRIPTION
**Pull request by denisdimitrov**
`experiences` were `undefined` because `venue.name` was `undefined` and the query was failing. `venueToUse` now is being used and `experiences` are fetched correctly. Types are also changed because the structure is different.

fixes https://github.com/sparkletown/internal-sparkle-issues/issues/199

### CODE REVIEW

File: src/components/molecules/TableHeader/TableHeader.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ const TableHeader: React.FC<TableHeaderProps> = ({
 }) => {
   const { user, profile } = useUser();
 
-  const experiences = useSelector(experiencesSelector);
+  const experience = useSelector(experiencesSelector);
```
Comment (0xdevalias) [inline src/components/molecules/TableHeader/TableHeader.tsx:25]: Why the change to experience (singular) here? The selector looks like it still refers to experiences (plural)?

File: src/components/molecules/TableHeader/TableHeader.tsx [hunk 2]
```diff
@@ -59,13 +59,13 @@ const TableHeader: React.FC<TableHeaderProps> = ({
       return false;
     }
     // Locked state is in the experience record
-    return experiences?.[venueName]?.tables?.[table]?.locked;
+    return experience?.tables?.[table]?.locked;
```
Comment (0xdevalias) [inline src/components/molecules/TableHeader/TableHeader.tsx:62]: Can you explain this? I'm guessing that at some point the structure of how we were keeping this data changed to no longer have the `venueName` in it like this?

File: src/components/molecules/TableHeader/TableHeader.tsx [hunk 3]
```diff
@@ -59,13 +59,13 @@ const TableHeader: React.FC<TableHeaderProps> = ({
...
   };
 
   const onLockedChanged = (tableName: string, locked: boolean) => {
     const doc = `experiences/${venueName}`;
     const update = {
-      tables: { ...experiences?.[venueName]?.tables, [tableName]: { locked } },
+      tables: { ...experience?.tables, [tableName]: { locked } },
```
Comment (0xdevalias) [inline src/components/molecules/TableHeader/TableHeader.tsx:68]: As above?

File: src/components/molecules/TablesUserList/TablesUserList.tsx [hunk 4]
```diff
@@ -71,11 +72,7 @@ const TablesUserList: React.FunctionComponent<PropsType> = ({
   const { user, profile } = useUser();
   const partygoers = usePartygoers();
   const isVenueUsersLoaded = useIsVenueUsersLoaded();
-  const { experience } = useSelector((state) => ({
-    experience:
-      state.firestore.data.experiences &&
-      state.firestore.data.experiences[venueName],
-  }));
+  const experience = useSelector(experiencesSelector);
```
Comment (0xdevalias) [inline src/components/molecules/TablesUserList/TablesUserList.tsx:75]: `experiencesSelector` doesn't seem to have `venueName` selection logic. Is this change equivalent? Or is the `venueName` part outdated/broken code?

```
export const experiencesSelector = (state: RootState) =>
  state.firestore.data.experiences;
```

File: src/types/Firestore.ts [hunk 5]
```diff
@@ -65,7 +65,7 @@ export interface FirestoreData {
   currentVenueNG?: AnyVenue;
   eventPurchase?: Record<string, Purchase>;
   events?: Record<string, VenueEvent>;
-  experiences: Record<string, Experience>;
+  experiences: Record<string, Record<string, Table>>;
```
Comment (0xdevalias) [inline src/types/Firestore.ts:68]: Can you please explain why this change? If we aren't using `Experience` here, should it be used anywhere anymore? What's the layout of this data structure?

**Comment by 0xdevalias** [comment]
@denisdimitrov https://github.com/sparkletown/internal-sparkle-issues/issues/199 specifically mentions 'conversation space' rather than 'jazz bar', which I believe are different templates (even though they share some underlying functionality). If they are different, do the changes here fix the 'conversation space' as well?

- `src/components/templates/ConversationSpace/ConversationSpace.tsx`
- `src/components/templates/Jazzbar/JazzTab/JazzTab.tsx`

File: src/components/molecules/TableHeader/TableHeader.tsx (hunk 2 above)
Comment (denisdimitrov) [reply src/components/molecules/TableHeader/TableHeader.tsx:62]: The query that fetches and storeAs gets the experience by venue name already. Which means what `experiences` is actually 1 single experience for your current venue.

File: src/components/molecules/TableHeader/TableHeader.tsx (hunk 3 above)
Comment (denisdimitrov) [reply src/components/molecules/TableHeader/TableHeader.tsx:68]: https://github.com/sparkletown/sparkle/pull/1094/files#r562731404

File: src/components/molecules/TablesUserList/TablesUserList.tsx (hunk 4 above)
Comment (denisdimitrov) [reply src/components/molecules/TablesUserList/TablesUserList.tsx:75]: https://github.com/sparkletown/sparkle/pull/1094/files#r562731404

**Comment by denisdimitrov** [comment]
@0xdevalias Your comments were taking an account, I applied the fix for Conversation Space as well and renamed the experiences to experience because it's singular atm.

File: src/components/templates/ConversationSpace/ConversationSpace.tsx [hunk 6]
```diff
@@ -7,6 +7,7 @@ import { currentVenueSelectorData } from "utils/selectors";
 import { useInterval } from "hooks/useInterval";
 import { useSelector } from "hooks/useSelector";
 import { usePartygoers } from "hooks/users";
+import { useExperiences } from "hooks/useExperiences";
```
Comment (0xdevalias) [inline src/components/templates/ConversationSpace/ConversationSpace.tsx:10]: We probably want to rename this to `useExperience` as a followup to this PR at some stage too.

File: src/components/templates/Jazzbar/JazzTab/JazzTab.tsx [hunk 7]
```diff
@@ -32,13 +31,13 @@ import { useSelector } from "hooks/useSelector";
...
 import { useVenueId } from "hooks/useVenueId";
 import { usePartygoers } from "hooks/users";
-import { useFirestoreConnect } from "hooks/useFirestoreConnect";
 
 import { addReaction } from "store/actions/Reactions";
 
 import { JAZZBAR_TABLES } from "./constants";
 
 import "./JazzTab.scss";
+import { useExperiences } from "hooks/useExperiences";
```
Comment (0xdevalias) [inline src/components/templates/Jazzbar/JazzTab/JazzTab.tsx:40]: Should be sorted with the rest of the `hooks/` imports.

**Review by 0xdevalias — COMMENTED** [review]
Retroactively approved, a few things we should clean up in a future PR.
